src/Toolkit/Toolkit.Maui/FeatureForm/BarcodeScanner.Android.cs

```diff
@@ -0,0 +1,210 @@
+            this.Touch += OnTap;
+            cameraController = new LifecycleCameraController(context)
+            {
+                ImageAnalysisBackpressureStrategy = ImageAnalysis.StrategyKeepOnlyLatest,
+                ImageAnalysisResolutionSelector = new AndroidX.Camera.Core.ResolutionSelector.ResolutionSelector.Builder().SetResolutionStrategy(
+                    new AndroidX.Camera.Core.ResolutionSelector.ResolutionStrategy(
+                    new Android.Util.Size(1920, 1080), AndroidX.Camera.Core.ResolutionSelector.ResolutionStrategy.FallbackRuleClosestHigherThenLower)).Build(),
+            };
+            cameraController.BindToLifecycle(lifecycleOwner);
+            previewView.Controller = cameraController;
+            barcodeImageAnalyzer = new BarcodeImageAnalyzer(OnBarcodeDetected);
+            cameraController.SetImageAnalysisAnalyzer(executor, barcodeImageAnalyzer);
+
+            barcodeFrame = new BarcodeFrame(context);
+            this.AddView(barcodeFrame, new LayoutParams(LayoutParams.MatchParent, LayoutParams.MatchParent));
+            barcodeFrame.BarcodeTapped += (s, e) => BarcodeTapped?.Invoke(this, e);
+        }
+
+        public event EventHandler<string>? BarcodeTapped;
+
+        private void OnBarcodeDetected(List<BarcodeInfo> barcodeInfoList)
+        {
+            barcodeFrame.Barcodes = barcodeInfoList;
+            barcodeFrame.Invalidate();
+            // set the auto-scan barcode if only one barcode is detected
+            if (barcodeInfoList.Count == 1)
+            {
+                // compare the current auto-scan barcode with the detected barcode to avoid setting
+                // the same barcode multiple times
+                if (autoScanBarcode?.RawValue != barcodeInfoList.First().RawValue)
+                {
+                    autoScanDelayCancellationToken?.Cancel();
+                    // once the auto-scan barcode is set, the delay routine will start
+                    autoScanBarcode = barcodeInfoList.First();
+                    autoScanDelayCancellationToken = new CancellationTokenSource();
+                    Task.Delay(AUTO_SCAN_DELAY_MS, autoScanDelayCancellationToken.Token).ContinueWith((t) =>
+                    {
+                        if (t.IsCompletedSuccessfully)
+                        {
+                            BarcodeTapped?.Invoke(this, autoScanBarcode.RawValue);
+                        }
+                    });
+                }
+            }
+            else
+            {
+                autoScanDelayCancellationToken?.Cancel();
+                autoScanBarcode = null;
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                autoScanDelayCancellationToken?.Dispose();
+                cameraController?.Dispose();
+                executor?.Dispose();
```

Comment:
Isn't this the main executor? Are we supposed to be disposing that?

This can probably be tested by opening/closing barcode scanner a couple times during the same app session, just to make sure we aren't accidentally disposing a long-lived shared object here.

Reply:
GetMainExecutor that creates the executor returns a new Java Object wrapper each time which must be disposed once we're done with it (it doesn't destroy the underlying java executor). Multiple scans are no issue.